src/collection/components/Cart/Cart.tsx

```diff
@@ -27,12 +28,43 @@ const Cart: React.FC<Props> = (props: Props) => {
       });
       return value;
     }
+
+    const handleItemClick = (item: CollectionItemModel) => {
+      if(item.tier !== CollectionItemTier.TIER1) return;
+      
+      const latestItem = cartList.find(x => x.collectionPhase === CollectionPhase.LATEST);
+      
+      if(latestItem?.tier !== CollectionItemTier.TIER1) return;
+
+      displayWarning();
+    }
+
+    const displayWarning = async () => {
+      setWarning(true);
+      let timer = setTimeout(() => setWarning(false), 5 * 1000);
+
+      // this will clear Timeout
+      // when component unmount like in willComponentUnmount
+      // and show will not change to true
+      return () => {
+        clearTimeout(timer);
+      };
```

Comment:
Is this return value needed? it seems that there is no usage.

Reply:
That means the timer cleared when the component unmounted.